sync_calendar.gs

```diff
@@ -265,4 +247,86 @@ function matchColor(status) {
 function getCountEventOfDay(cur_data) {
   let count = cur_data.split('').filter(char => char === '@').length;
   return count;
+}
+
+function isTimeOut(today) {
+  var now = new Date();
+  return now.getTime() - today.getTime() > 28000;
+}
+
+const _insertDashboardLink = (sheet, editedRow, editedValue) => {
+  // Specify the column number where you want to add the link
+  const linkColumn = 18; // For example, column G
+
+  // Get the cell to which you want to add the link
+  const linkCell = sheet.getRange(editedRow, linkColumn);
+
+  // Construct the link URL based on the edited value
+  // it will be replaced with the actual logic to construct the link URL
+  const linkUrl = `https://metabase.pycon.tw/question/49-pycon-taiwan`
+
+  // Create a rich text value with the link
+  const richTextValue = SpreadsheetApp.newRichTextValue()
+    .setText(editedValue)
+    .setLinkUrl(linkUrl)
+    .build();
+
+  // Set the rich text value to the link cell
+  linkCell.setRichTextValue(richTextValue);
+}
+
+const _updateScheduledPosts = (range, sheet, row) => {
+  let startTime = new Date();
+  let editRange = range.getA1Notation().split(":");
+
+  var editState = sheet.getRange(1, 23).getValue();
+  while(editState == "更新中") {
+    editState = sheet.getRange(1, 23).getValue();
+
+    if(isTimeOut(startTime)) {
+      console.log("Time Out");
+      sheet.getRange(1, 23).setValue("--");
+      return;
+    }
+  }
+
+  // 根據 編輯範圍 採取對應處理方式
+  if(editRange.length > 1) {
+    console.log("Start: "+ editRange[0] + ",End: "+ editRange[1]);
+
+    // 編輯 多欄位，且欄位範圍符合條件則採取 刷新日曆
+    sheet.getRange(1, 23).setValue("更新中");
+    reflashCalendar();
+    sheet.getRange(1, 23).setValue("--");
+  } else {
+    console.log("Place: "+ editRange[0].charCodeAt(0));
+
+    // 編輯 單一欄位，且欄位範圍符合條件則採取 刷新日曆
+    if(checkCellPlace(editRange[0].charCodeAt(0))) {
+      sheet.getRange(1, 23).setValue("更新中");
+      let data = getRowData(sheet, row);
```

Comment:
@winniepopu need to have your eyes on this~ Actually I didn't change anything but put them into this func

Reply:
These changes are to fix the timeout issues during updates. Everything has been checked and looks good.

Thanks, @david30907d & @winniepopu